### ISSUE TITLE
Segmentation fault on latest release

### ISSUE THREAD

**Issue by ghost** [twistedfall/opencv-rust#314]
```rust
use opencv::core::{Point, Vector};
use opencv::imgproc::{bounding_rect, min_area_rect};
use opencv::prelude::*;

fn main() {
    unsafe {
        let mut contour = Vector::<Point>::new();
        let points = vec![
            // Point::new(0, 0),
            // Point::new(0, 10),
            // Point::new(10, 10),
            // Point::new(10, 0),
            // Point::new(0, 0),
            Point::new(5, 5),
            Point::new(5, 15),
            Point::new(15, 15),
            Point::new(15, 5),
            Point::new(5, 5),
            // Point::new(0, 0),
        ];
        for point in points {
            contour.push(point);
        }
        dbg!(bounding_rect(&contour).unwrap());
        let t = min_area_rect(&contour).unwrap();
        dbg!(t));
    }
}
```

This program segfaults on the last line when using latest release. Works fine on `0.56`.

1. Operating system
    popos
2. The way you installed OpenCV: package, official binary distribution, manual compilation, etc.
    apt install libopencv-dev
3. OpenCV version
    4.5.1
4. rustc version (`rustc --version`)
    1.57.0

I'll attach build logs soon.

**Comment by twistedfall** [comment]
I cannot immediately confirm this as it works without a segfault on my dev system (Gentoo, OpenCV 4.5.5, rustc 1.57). I'll try to reproduce under your specific circumstances.

**Comment by twistedfall** [comment]
The reason for this issue is the mismatch between how Rust expects C calling convention to return a `struct` and how the default C++ compiler in Ubuntu 20.04 actually does. I'll rework this internally to work around the issue

**Comment by twistedfall** [comment]
This issue should now be fixed in v0.62
